### PR TITLE
Settings first option implemented(change downloads directory)

### PR DESCRIPTION
**Pull request by amoghbl1**
settings change downloads directory functionality added.

### CODE REVIEW

File: .classpath [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
-	<classpathentry kind="con" path="com.android.ide.eclipse.adt.ANDROID_FRAMEWORK"/>
+	<classpathentry exported="true" kind="con" path="com.android.ide.eclipse.adt.ANDROID_FRAMEWORK"/>
```
Comment (phinfinity) [inline .classpath]: I'm not sure what this does, but I think this isn't relevant to this commit (remove?)

File: src/com/phinmadvader/andcpp/Constants.java [hunk 2]
```diff
@@ -15,4 +16,15 @@
     public static int DOWNLOAD_TIMEOUT_MILLIS = 10000;
     public static int DOWNLOAD_UPDATE_INTERVAL_MILLIS = 100; // Updates all download activities every 200ms
     public static int POLLING_CLASS_TIME_IN_MILLIS = 250; // Interval frequency for poll updates
+    
+    // Helper functions.
+    public static String getDCDirectory(){
```
Comment (phinfinity) [inline src/com/phinmadvader/andcpp/Constants.java]: Move these changes to Settings.java ? Would prefer to keep this class only with static initialized constant values.
(Make sure there aren't any stray changes to this file)
If you think it makes it more understandable you cna rename dcdirectory to defaultdcdirectory or something like that

File: src/com/phinmadvader/andcpp/MainActivity.java [hunk 3]
```diff
@@ -5,6 +5,7 @@
 import android.app.ActionBar;
 import android.app.ProgressDialog;
 import android.content.BroadcastReceiver;
+import android.content.ClipData.Item;
```
Comment (phinfinity) [inline src/com/phinmadvader/andcpp/MainActivity.java]: huh ? review your diff to remove stray changes like these

File: src/com/phinmadvader/andcpp/MainActivity.java [hunk 4]
```diff
@@ -120,6 +123,9 @@ public SharedPreferences getprefs() {
 	@Override
 	public void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
+		
+		context = getApplicationContext();
```
Comment (phinfinity) [inline src/com/phinmadvader/andcpp/MainActivity.java]: is this context variable actually being used somewhere? (if not remove it), also I think context here should only be "this" (MainActivity itself)

File: .classpath (hunk 1 above)
Comment (amoghbl1) [inline .classpath]: Ya, I'll remove it... It looks like something eclipse changed... And add of now, the directory needs to be entered directly, but I guess selecting a directory(file browser) would be easier! That's a little harder to implement and I'll try to get that done next...

Comment (phinfinity) [inline .classpath]: No problem, text entry should be good enough as of now, can implement directory selection later in another commit

File: src/com/phinmadvader/andcpp/Settings.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+public class Settings extends Activity{
+	
+	private EditText inputDirectoryTextView;
+	
+	@Override
+	public void onCreate(Bundle savedInstanceState){
+		super.onCreate(savedInstanceState);
+		
+		//although it's called file list, it's the exact code needed here, so I'm not going to make a new view.
+		setContentView(R.layout.file_list);
```
Comment (phinfinity) [inline src/com/phinmadvader/andcpp/Settings.java:27]: Settings looks a bit bland, looks like just a random bit of text, wouldn't have expected it to be clickable from the look.
Maybe use a different layout?
Mainly a divider line would be nice to have I think.

**Comment by phinfinity** [comment]
Core functionality looks good! , I've added some small comments which need to be fixed.

**Comment by amoghbl1** [comment]
Ya, I'll implement all those and get it ready by the tomorrow night...

File: src/com/phinmadvader/andcpp/MainActivity.java (hunk 3 above)
Comment (amoghbl1) [inline src/com/phinmadvader/andcpp/MainActivity.java]: removed, I've also removed some unused imports in other files...

File: src/com/phinmadvader/andcpp/MainActivity.java (hunk 4 above)
Comment (amoghbl1) [inline src/com/phinmadvader/andcpp/MainActivity.java]: removed this, modified the getDCDownloadsDirectory function to accept a context...

**Comment by amoghbl1** [comment]
Ready for review... changed the layout for the settings list items, look kinda better now but you can change it to look however you want, also moved the functions out of Constants and into Settings.

**Comment by phinfinity** [comment]
I've made some changes to your commit and merged it as 6e1e12fb467b04 : 
- you missed a place where Constants.dcDirectory was still being used (SearchResultsFragment)
- I've moved the folder creation code from MainActivity into your Settings so that folder is created. Otherwise it would have caused a bug when settings is changed after connecting

Layout looks good :+1: 

Additionally I've also made commit 02956ce4 , which sets the default value of the dialog box to current value so that it is easier to type the path
and commit 1dd6a925cd3 , forces re-rendering the directory path shown in login fragment to reflect updated directory path

**Pull request closed without merging**